tensorflow_probability/examples/jupyter_notebooks/Deep_Exponential_Family_TFP.ipynb

```diff
@@ -0,0 +1,1904 @@
+        "    w2 = tfd.Gamma(tf.fill([units[2], units[1]], 0.1), tf.fill([units[2], units[1]], 0.3), name=\"w2\")\n",
+        "    w1 = tfd.Gamma(tf.fill([units[1], units[0]], 0.1), tf.fill([units[1], units[0]], 0.3), name=\"w1\")\n",
+        "    w0 = tfd.Gamma(tf.fill([units[0], feature_size], 0.1), tf.fill([units[0], feature_size], 0.3), name=\"w0\")\n",
+        "\n",
+        "    z2 = tfd.Gamma(tf.fill([data_size, units[2]], 0.1), tf.fill([data_size, units[2]], 0.1), name=\"z2\")\n",
+        "    z1 = tfd.Gamma(shape, shape / tf.matmul(z2.sample(), w2.sample()), name=\"z1\")\n",
+        "    z0 = tfd.Gamma(shape, shape / tf.matmul(z1.sample(), w1.sample()), name=\"z0\")\n",
+        "    x = tfd.Poisson(tf.matmul(z0.sample(), w0.sample()), name=\"x\")\n",
+        "    return x\n"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "FF1TvorGnbpP",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "The first of these is our learnable deterministic distribution over positive reals."
+      ]
+    },
+    {
+      "metadata": {
+        "id": "S3ubbvHGCDrF",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def trainable_positive_deterministic(shape, min_loc=1e-3, name=None):\n",
+        "    \"\"\"Learnable Deterministic distribution over positive reals.\"\"\"\n",
+        "    with tf.variable_scope(None, default_name=\"trainable_positive_deterministic\"):\n",
+        "        unconstrained_loc = tf.get_variable(\"unconstrained_loc\", shape)\n",
+        "        loc = tf.maximum(tf.nn.softplus(unconstrained_loc), min_loc)\n",
+        "        rv = tfd.Deterministic(loc=loc, name=name)\n",
+        "        return rv"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "metadata": {
+        "id": "d-W1oPqnndUG",
+        "colab_type": "text"
+      },
+      "cell_type": "markdown",
+      "source": [
+        "Next up we have the learnable gamma distribution ($\\text{Gamma}(\\alpha, \\beta)$) via concentration and scale parameterization. We have a gamma distribution. "
+      ]
+    },
+    {
+      "metadata": {
+        "id": "pv6-PWqfCMDl",
+        "colab_type": "code",
+        "colab": {}
+      },
+      "cell_type": "code",
+      "source": [
+        "def trainable_gamma(shape, min_concentration=1e-3, min_scale=1e-5, name=None):\n",
+        "    \"\"\"Learnable Gamma via concentration and scale parameterization.\"\"\"\n",
+        "    with tf.variable_scope(None, default_name=\"trainable_gamma\"):\n",
+        "        unconstrained_concentration = tf.get_variable(\n",
+        "            \"unconstrained_concentration\", shape,\n",
+        "            initializer=tf.random_normal_initializer(mean=0.5, stddev=0.1))\n",
+        "        unconstrained_scale = tf.get_variable(\n",
```

Comment:
the v2 way would be to explicitly init a Variable instance
`tf.compat.v2.Variable(..)`